File: frontend/src/services/noteService.js
import api from './api';

export const noteService = {
  getUserNotes: async (articleId = null) => {
    const params = new URLSearchParams();
    if (articleId) params.append('articleId', articleId);

    const response = await api.get(`/notes?${params.toString()}`);
    return response.data;
  },

  createNote: async (noteData) => {
    const response = await api.post('/notes', noteData);
    return response.data;
  },

  deleteNote: async (noteId) => {
    const response = await api.delete(`/notes/${noteId}`);
    return response.data;
  },
};

File: backend/src/main/java/com/readwise/ai/modules/notes/service/NoteService.java
package com.readwise.ai.modules.notes.service;

import com.readwise.ai.modules.notes.dto.CreateNoteRequest;
import com.readwise.ai.modules.notes.dto.NoteDto;

import java.util.List;

public interface NoteService {
    List<NoteDto> getUserNotes(String userEmail, Long articleId);
    NoteDto createNote(String userEmail, CreateNoteRequest request);
    void deleteNote(String userEmail, Long noteId);
}
